frontend/packages/dev-console/src/components/pipelines/pipeline-form/SecretsList.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ResourceLink, Firehose, FirehoseResult } from '@console/internal/components/utils';
+import { SecretModel } from '@console/internal/models';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecondaryStatus } from '@console/shared';
+import { SecretKind } from '@console/internal/module/k8s';
+import './SecretsList.scss';
+
+type SecretsProps = {
+  secrets?: FirehoseResult<SecretKind[]>;
+};
+
+type SecretsListProps = {
+  namespace: string;
+};
+
+const secretTypes = [SecretType.dockerconfigjson, SecretType.basicAuth, SecretType.sshAuth];
+
+const Secrets: React.FC<SecretsProps> = ({ secrets }) => {
+  const filterData = _.filter(secrets.data, (secret) => _.includes(secretTypes, secret.type));
+
+  return (
+    <div className="odc-secrets-list">
+      <label>Source Secrets ({filterData.length})</label>
+      <div className="odc-secrets-list__secrets">
+        {!_.isEmpty(filterData) ? (
+          filterData.map((secret) => {
```

Comment:
This is an unnecessary check.

```
_.filter(null, (v) => v) // []
```
Filter always returns an array. And `.map` won't execute with an empty array.